### PR TITLE
Integration documentation reworked

### PR DESCRIPTION
**Pull request by comradekingu**
Attempt at fixing issue #5009

### CODE REVIEW

File: docs/devel/integration.rst [hunk 1]
```diff
@@ -55,17 +58,22 @@ translations to match your code base.
...
-likely have existing workflow for developers to introduce new strings. You
-might extend it by using :ref:`source-quality-gateway`.
+In case your translation files are stored in VCS together with the code,
+you most likely have existing workflow for developers to introduce new strings.
+You might extend it by using 
+
+Any way of adding strings will be picked up, but consider using
+:ref:`source-quality-gateway` to avoid also introducing errors.
 
 When the translation files are separate, there needs to be a way to introduce
```
Comment (comradekingu) [inline docs/devel/integration.rst:68]: separate from?

Comment (nijel) [reply docs/devel/integration.rst:68]: Not sure ... it probably should have been standalone. This paragraph talks about having translation files not in the VCS repository with the code and managing it externally.

Comment (comradekingu) [reply docs/devel/integration.rst:68]:
```suggestion
When the translation files are not in a upstream repository, there needs to be a way to introduce
```

File: docs/devel/integration.rst [hunk 2]
```diff
@@ -6,47 +6,50 @@ Integrating with Weblate
...
 
-Pushing string changes to Weblate
-+++++++++++++++++++++++++++++++++
+Weblate represents a branch from your VCS upstream that you can fetch changed
+strings from just like any other.
 
-To push newly updated strings to Weblate, just let it pull from the upstream
-repository. This can be achieved in the user interface (in the :guilabel:`Repository
-maintenance`) or from command line using :ref:`wlc`.
+Either fetch the filesystem, or use :ref:`git-exporter`). Commit any pending
```
Comment (nijel) [inline docs/devel/integration.rst:46]: Closing bracket should not be here:

```suggestion
Either fetch the filesystem, or use :ref:`git-exporter`. Commit any pending
```

File: docs/devel/integration.rst [hunk 3]
```diff
@@ -6,47 +6,50 @@ Integrating with Weblate
...
 
 .. seealso::
 
    :ref:`adding-projects`,
    :ref:`faq-submodule`
 
-Getting translations updates from Weblate
-+++++++++++++++++++++++++++++++++++++++++
+Send string changes to Weblate
++++++++++++++++++++++++++++++++++
```
Comment (nijel) [inline docs/devel/integration.rst:23]:
```suggestion
Send string changes to Weblate
++++++++++++++++++++++++++++++
```

File: docs/devel/integration.rst [hunk 4]
```diff
@@ -55,17 +58,22 @@ translations to match your code base.
 Adding new strings
 ++++++++++++++++++
 
-In case your translation files are stored in VCS together with the code, you most
-likely have existing workflow for developers to introduce new strings. You
-might extend it by using :ref:`source-quality-gateway`.
+In case your translation files are stored in VCS together with the code,
+you most likely have existing workflow for developers to introduce new strings.
+You might extend it by using
```
Comment (nijel) [inline docs/devel/integration.rst:63]: Incomplete sentence?

File: docs/devel/integration.rst [hunk 5]
```diff
@@ -55,17 +58,22 @@ translations to match your code base.
...
+You might extend it by using 
+
+Any way of adding strings will be picked up, but consider using
+:ref:`source-quality-gateway` to avoid also introducing errors.
 
-When the translation files are separate, there needs to be a way to introduce
-new strings. Weblate can add new strings on monolingual translations only (see
-:ref:`bimono`). You have three options to do that:
+When the translation files are not in a upstream repository, there needs to be a way to introduce
+new strings.
```
Comment (nijel) [inline docs/devel/integration.rst:69]: The link to bi/mono-lingual format is gone...

File: docs/devel/integration.rst [hunk 6]
```diff
@@ -55,17 +58,22 @@ translations to match your code base.
...
 * Manually, using :guilabel:`Add new translation string` from :guilabel:`Tools`
-  menu on source language translation.
-* Programatically, using API :http:post:`/api/translations/(string:project)/(string:component)/(string:language)/units/`.
-* By uploading source file as :guilabel:`Replace existing translation file`
-  (this overwrites existing strings, so please make sure the file includes both
-  old and new ones, see :ref:`upload-method`).
+  menu on the translation language used as the source for others.
+  (Only works for monolingual translations :ref:`bimono`)
+* Alternatively, you can use the API
+:http:post:`/api/translations/(string:project)/(string:component)/(string:language)/units/`.
```
Comment (nijel) [inline docs/devel/integration.rst:75]: Missing indentation: 

```suggestion
  :http:post:`/api/translations/(string:project)/(string:component)/(string:language)/units/`.
```

File: docs/devel/integration.rst [hunk 7]
```diff
@@ -55,17 +58,22 @@ translations to match your code base.
...
+* It is also possible to just upload a new source file as
+:guilabel:`Replace existing translation file`
+ (this overwrites existing strings, so make sure the file includes both old
```
Comment (nijel) [inline docs/devel/integration.rst:78]: Missing indentation:

```suggestion
  :guilabel:`Replace existing translation file`
  (this overwrites existing strings, so make sure the file includes both old
```

File: docs/devel/integration.rst (hunk 5 above)
Comment (comradekingu) [reply docs/devel/integration.rst:67]: It is a bit farther down. I seem to have thought there are different rules for the API…

Comment (nijel) [reply docs/devel/integration.rst:67]: It is not related to the API, but is general Weblate limitation (for now).

File: docs/devel/integration.rst [hunk 8]
```diff
@@ -40,16 +40,17 @@ translations to match the latest version of what it is you want to translate.
...
 
 Either fetch the filesystem, or use :ref:`git-exporter`. Commit any pending
 changes first (see :ref:`lazy-commit`) from the user interface
 (in the :guilabel:`Repository maintenance`) or from the command-line
 using :ref:`wlc`.
 
-Automation is possible by granting the Weblate user push access to your
-repository and configuring :ref:`component-push` in the :ref:`component`.
+Automation is possible by granting the Weblate user (available for various
+version control systems) push access to your repository and configuring
```
Comment (nijel) [inline docs/devel/integration.rst:52]:
```suggestion
Automation is possible by granting the Weblate user (available on various
code hosting sites) push access to your repository and configuring
```

Comment (comradekingu) [reply docs/devel/integration.rst:52]: I even read this the wrong way around to comply with what I thought I wrote ;)

File: docs/devel/integration.rst [hunk 9]
```diff
@@ -6,47 +6,50 @@ Integrating with Weblate
 Importing localization project into Weblate
 +++++++++++++++++++++++++++++++++++++++++++
 
-Weblate was developed with VCS integration in mind. The easiest approach to
-integrate with it is to grant access to your VCS repository. The import process
-will guide you through configuring components with your translations.
+You can either use a version-control system to hold your translations,
+by granting Weblate access to it, and follow the guide to configure your
```
Comment (comradekingu) [inline docs/devel/integration.rst:10]:
```suggestion
by granting Weblate access to it, and letting it guide you through configuring your
```

**Review by jace4511 — APPROVED** [review]

**Review by jace4511 — APPROVED** [review]

**Review by jace4511 — APPROVED** [review]

**Review by jace4511 — APPROVED** [review]

File: docs/devel/integration.rst [hunk 10]
```diff
@@ -55,17 +58,22 @@ translations to match your code base.
...
+  menu on the translation language used as the source for others.
+  (Only works for monolingual translations :ref:`bimono`)
+* Alternatively, you can use the API
+  :http:post:`/api/translations/(string:project)/(string:component)/(string:language)/units/`.
+  (Only works for monolingual translations :ref:`bimono`)
+* It is also possible to just upload a new source file as
+  :guilabel:`Replace existing translation file`
+  (this overwrites existing strings, so make sure the file includes both old
+  and new ones, see :ref:`upload-method`).
+
```
Comment (nijel) [inline docs/devel/integration.rst:79]: Lint fix:

```suggestion
```

**Comment by nijel** [comment]
@comradekingu Not sure if you've noticed that, but there is now Read the Docs building each pull request, and you can browse the resulting docs online to review your changes. The link is always present in the status checks (docs/readthedocs.org:weblate).

This pull request can be viewed here: https://weblate--5097.org.readthedocs.build/en/5097/devel/integration.html

File: docs/devel/integration.rst [hunk 11]
```diff
@@ -6,47 +6,50 @@ Integrating with Weblate
...
+This can be automated by installing a webhook on your repository to trigger
+Weblate whenever there is a new commit, see :ref:`update-vcs` for more details.
+
+Alternateively, you can use the UI or :doc:`/api` to update translations
+to match the latest version of what it is you want to translate.
 
 .. seealso::
 
     :ref:`continuous-translation`
+
```
Comment (nijel) [inline docs/devel/integration.rst:38]: Another lint issue: 

```suggestion

```

File: docs/devel/integration.rst [hunk 12]
```diff
@@ -6,47 +6,49 @@ Integrating with Weblate
...
+This can be automated by installing a webhook on your repository to trigger
+Weblate whenever there is a new commit, see :ref:`update-vcs` for more details.
+
+Alternateively, you can use the UI or :doc:`/api` to update translations
+to match the latest version of what it is you want to translate.
 
 .. seealso::
 
     :ref:`continuous-translation`
+Fetch changes made in Weblate
```
Comment (nijel) [inline docs/devel/integration.rst:38]:
```suggestion

Fetch changes made in Weblate
```

File: docs/devel/integration.rst (hunk 5 above)
Comment (orangesunny) [reply docs/devel/integration.rst:67]: I rewrote this again as all three options are monolingual only.

**Review by orangesunny — APPROVED** [review]
thanks for the contribution, @comradekingu!

Please, be also more careful next time.

File: docs/devel/integration.rst [hunk 13]
```diff
@@ -6,46 +6,48 @@ Integrating with Weblate
...
+all the translations without integration.
 
 .. seealso::
 
    :ref:`adding-projects`,
    :ref:`faq-submodule`
 
-Getting translations updates from Weblate
+Fetching updated translations from Weblate
 +++++++++++++++++++++++++++++++++++++++++
```
Comment (nijel) [inline docs/devel/integration.rst:22]:
```suggestion
++++++++++++++++++++++++++++++++++++++++++
```

File: docs/devel/integration.rst [hunk 14]
```diff
@@ -6,46 +6,48 @@ Integrating with Weblate
...
+to match their latest version.
+
 .. seealso::
 
     :ref:`continuous-translation`
 
-Pushing string changes to Weblate
-+++++++++++++++++++++++++++++++++
+Fetching remote changes into Weblate
++++++++++++++++++++++++++++++
```
Comment (nijel) [inline docs/devel/integration.rst:41]:
```suggestion
++++++++++++++++++++++++++++++++++++
```

File: docs/devel/integration.rst [hunk 15]
```diff
@@ -6,46 +6,48 @@ Integrating with Weblate
 Importing localization project into Weblate
 +++++++++++++++++++++++++++++++++++++++++++
 
-Weblate was developed with VCS integration in mind. The easiest approach to
-integrate with it is to grant access to your VCS repository. The import process
-will guide you through configuring components with your translations.
+Weblate has been developed with VCS integration in mind as it’s core feature, so the easiest way is
```
Comment (nijel) [inline docs/devel/integration.rst:9]:
```suggestion
Weblate has been developed with VCS integration in mind as it’s core feature, so the easiest way is
```

File: docs/devel/integration.rst [hunk 16]
```diff
@@ -6,46 +6,48 @@ Integrating with Weblate
...
+to grant Weblate the access to your repository.
```
Comment (nijel) [inline docs/devel/integration.rst:10]:
```suggestion
to grant Weblate the access to your repository.
```

**Pull request merged**